docs/userguide.md

```diff
@@ -0,0 +1,192 @@
++ It utilizes several strategies to push down the computing to reduce the size of dataset handling by Spark SQL, which accelerates the query execution. It also uses the TiDB built-in statistical information for  the query plan optimization.
++ From the data integration point of view, TiSpark + TiDB provides a solution runs both transaction and analysis directly on the same platform without building and maintaining any ETLs. It simplifies the system architecture and reduces the cost of maintenance.
++ In addition, you can deploy and utilize tools from the Spark ecosystem for further data processing and manipulation on TiDB. For example, using TiSpark for data analysis and ETL; retrieving data from TiKV as a machine learning data source; generating reports from the scheduling system  and so on.
+
+## Environment Setup
+
++ The current version of TiSpark supports Spark 2.1. For Spark 2.0 and Spark 2.2, it has not been fully tested yet . It does not support any versions earlier than 2.0.
++ TiSpark requires JDK 1.8+ and Scala 2.11 (Spark2.0 + default Scala version).
++ TiSpark runs in any Spark mode such as YARN, Mesos, and Standalone.
+
+
+## Recommended configuration
+
+### Deployment of TiKV and TiSpark clusters
+
+#### Configuration of the TiKV cluster
+
+For independent deployment of TiKV and TiSpark, it is recommended to refer to the following recommendations
+ 
++ Hardware configuration
+ - For general purposes, please refer to the TiDB and TiKV hardware configuration [recommendations](https://github.com/pingcap/docs/blob/master/op-guide/recommendation.md#deployment-recommendations).
+ - If the usage is more focused on the analysis scenarios, you can increase the memory of the TiKV nodes to at least 64GIf using  Hard Disk Drive (HDD), it is recommended to use at least 8 disks.
+
++ TiKV parameters (default)
+
+```
+[Server]
+End-point-concurrency = 8 # For OLAP scenarios, consider increasing this parameter
+[Raftstore]
+Sync-log = false
+
+
+[Rocksdb]
+Max-background-compactions = 6
+Max-background-flushes = 2
+
+[Rocksdb.defaultcf]
+Block-cache-size = "10GB"
+
+[Rocksdb.writecf]
+Block-cache-size = "4GB"
+
+[Rocksdb.raftcf]
+Block-cache-size = "1GB"
+
+[Rocksdb.lockcf]
+Block-cache-size = "1GB"
+
+[Storage]
+Scheduler-worker-pool-size = 4
+```
+
+#### Configuration of the independent deployment of the Spark cluster and the TiSpark cluster 
+
+ 
+Please refer to the [Spark official website](https://spark.apache.org/docs/latest/hardware-provisioning.html) for the detail hardware recommendations.
+
+The following is a short overview of the TiSpark configuration.
+
+Generally, it is  recommended to allocat 32G memory for Spark. Please reserve at least 25% of the memory for the operating system and buffer cache.
+
+It is recommended to provision at least 8 to 16 cores on per machine for Spark. Initially, you can assign all the CPU cores to Spark.
```

Comment:
Please refer to Spark standalone configuration.
https://spark.apache.org/docs/latest/hardware-provisioning.html